### PR TITLE
Adding more emails, moving tabs logic into context

### PR DESCRIPTION
**Pull request by philyeung**

### CODE REVIEW

File: src/pages/MainPage.jsx [hunk 1]
```diff
@@ -2,49 +2,52 @@ import { useEffect, useState } from "react";
...
+   //   .catch((err) => console.log(err));
+
+    // setMovies(moviesData.filter())
   }, [search]);
 
   return (
-    <div>
-      <div className="w-full flex flex-col">
+  <div>
+    <NavBar />
```
Comment (KushTurner) [inline src/pages/MainPage.jsx]: I think you forgot to remove this.

File: src/components/MovieCard.jsx [hunk 2]
```diff
@@ -29,27 +29,20 @@ export default function MovieCard(props) {
...
-        payload: { favouriteMovies: [...favouriteMovies, props.movie] },
-      });
-
-      MovieService.getRecommendedMovies(imdbID)
+    dispatch({
+        type: Actions.UpdateFavourites,
+        payload: props.movie
+     })
+    
+    MovieService.getRecommendedMovies(imdbID)
```
Comment (KushTurner) [inline src/components/MovieCard.jsx]: Should we get rid of the recommended tab? We are no longer including the AI/ML task. This was discussed in the call on Tuesday.

We could keep the tab (for looks) but just remove this part as it isn't needed.

File: inbox/email_9.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+Can we add a search to the top of the page please? We think the first thing that users want to do is just to be able to search quickly for the film they are looking for
+
+To start with, can we just take a look at what it looks like on the page?
+
+Phil
+
+## Tips
+
+There's already a pre-made [Search](../src/components/SearchBar.jsx) in the Components folder. Can you see how you would import it into the layout of the website?
```
Comment (KushTurner) [inline inbox/email_9.md:11]: FYI, search has 2 parameters setSearch and search, as discussed for the NavBar it's not intuitive to add that without knowing React.

https://github.com/nfeagh/react-flask-app/blob/main/src/components/SearchBar.jsx#L8
https://github.com/nfeagh/react-flask-app/blob/main/src/components/SearchBar.jsx#L10

Could we just mention how to add the arguments in this email or do something else?

File: inbox/email_9B.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+## Tips
+
+On the main page, you will see a function that is commented out:
+
+`setMovies(moviesData.filter())`
+
+If this code is commented out, it will run everytime something is typed into the search bar. The filter method takes a function and runs it against the array of movies. To find out more, look [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter)
+
+Can you create a function, that filters through the list of movies, but only keeps the movies that contain the search term entered in the search bar?
```
Comment (KushTurner) [inline inbox/email_9B.md:15]: I really like this idea, I am going to add more mock movies so it's not just star wars because of this.

**Review by KushTurner — CHANGES_REQUESTED** [review]

File: src/pages/MainPage.jsx (hunk 1 above)
Comment (philyeung) [reply src/pages/MainPage.jsx]: Whoops - thanks!

File: inbox/email_9.md (hunk 3 above)
Comment (philyeung) [reply inbox/email_9.md:11]: I did completely forget that - and weirdly enough it works without passing in the search parameter! The functionality works fine for me if I just pass it the setSearch function... I've amended the emails though to make it clear it needs a function passed in

**Comment by philyeung** [comment]
By the way, for full transparency, the way that I've added a "movie" as a payload in the add/remove favourites isn't great (the payload suddenly changing isn't intuitive and would break in Typescript), but just making do for the time being just as an intial refactor

**Comment by KushTurner** [comment]
Feel free to merge this. Thanks for the changes. 

> By the way, for full transparency, the way that I've added a "movie" as a payload in the add/remove favourites isn't great (the payload suddenly changing isn't intuitive and would break in Typescript), but just making do for the time being just as an intial refactor

ye, but the initial design I made wasn't the best anyways so we kinda have to make do.

**Review by KushTurner — APPROVED** [review]

**Pull request merged**